src/misc/cache/class/lv_cache_sc_da.c

```diff
@@ -0,0 +1,507 @@
+                             void ** cache_data, lv_cache_entry_t ** cache_entry);
+
+/**********************
+ *  GLOBAL VARIABLES
+ **********************/
+const lv_cache_class_t lv_cache_class_sc_da = {
+    .alloc_cb = alloc_cb,
+    .init_cb = init_cnt_cb,
+    .destroy_cb = destroy_cb,
+
+    .get_cb = get_cb,
+    .add_cb = add_cb,
+    .remove_cb = remove_cb,
+    .drop_cb = drop_cb,
+    .drop_all_cb = drop_all_cb,
+    .get_victim_cb = get_victim_cb,
+    .reserve_cond_cb = reserve_cond_cb,
+    .iter_create_cb = cache_iter_create_cb,
+};
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+#define LV_CACHE_ENTRY_SIZE lv_cache_entry_get_size(0)
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+/**********************
+ *   STATIC FUNCTIONS
+ **********************/
+
+static inline void get_entry(lv_cache_sc_da_t * da, size_t index,
+                             void ** cache_data, lv_cache_entry_t ** cache_entry)
+{
+    const size_t node_size = lv_cache_entry_get_size(da->cache.node_size);
+    *cache_data = da->data + (index * node_size);
+    *cache_entry =
+        lv_cache_entry_get_entry(*cache_data, da->cache.node_size);
+}
+static inline void set_second_chance(lv_cache_entry_t * entry, bool value)
+{
+    lv_cache_entry_set_class_flag(entry, value);
+}
+
+static inline bool get_second_chance(lv_cache_entry_t * entry)
+{
+    return lv_cache_entry_get_class_flag(entry);
+}
+
+static void * alloc_new_entry(lv_cache_sc_da_t * da, const void * key,
+                              void * user_data)
+{
+    LV_UNUSED(user_data);
+
+    LV_ASSERT_NULL(da);
+    LV_ASSERT_NULL(key);
+
+    if(da == NULL || key == NULL) {
+        return NULL;
+    }
+    const size_t node_size = lv_cache_entry_get_size(da->cache.node_size);
+
+    if(da->capacity == da->cache.size) {
+        if(da->capacity == da->cache.max_size) {
+            LV_LOG_ERROR(
+                "Reached maximum size of cache. Unable to allocate a new entry");
+            return NULL;
+        }
+
+        const size_t new_capacity =
+            LV_MIN(da->capacity == 0 ? 1 : da->capacity * 2,
+                   da->cache.max_size);
+
+        uint8_t * new_data = (uint8_t *)lv_realloc(
+                                 da->data, new_capacity * node_size);
+
+        if(!new_data) {
+            LV_LOG_ERROR("Failed to allocate new data for cache");
+            return NULL;
+        }
+
+        da->data = new_data;
+        da->capacity = new_capacity;
+    }
+    void * last_da_entry;
+    lv_cache_entry_t * last_cache_entry;
+
+    get_entry(da, da->cache.size, &last_da_entry, &last_cache_entry);
+
+    lv_memcpy(last_da_entry, key, da->cache.node_size);
+    lv_cache_entry_init(last_cache_entry, &da->cache, da->cache.node_size);
+    lv_cache_entry_disable_deleting(last_cache_entry);
```

Comment:
I don't understand something: `disable_deleting` is set here when a new entry is allocated, but when this flag is cleared to allow deleting?

Reply:
Ahh, I see.